platform/cxxsupport/mstd_functional

```diff
@@ -0,0 +1,508 @@
+    template <typename T, typename U>
+    constexpr auto operator()(T&& t,U&& u) const -> decltype(std::forward<T>(t) | std::forward<U>(u))
+    {
+        return std::forward<T>(t) | std::forward<U>(u);
+    }
+};
+
+// [bitwise.operations.xor]
+template <typename T = void>
+struct bit_xor {
+    constexpr T operator()(const T &x, const T &y) const
+    {
+        return x ^ y;
+    }
+};
+
+template <>
+struct bit_xor<void> {
+    using is_transparent = true_type;
+    template <typename T, typename U>
+    constexpr auto operator()(T&& t,U&& u) const -> decltype(std::forward<T>(t) ^ std::forward<U>(u))
+    {
+        return std::forward<T>(t) ^ std::forward<U>(u);
+    }
+};
+
+// [bitwise.operations.not]
+template <typename T = void>
+struct bit_not {
+    constexpr T operator()(const T &arg) const
+    {
+        return ~arg;
+    }
+};
+
+template <>
+struct bit_not<void> {
+    using is_transparent = true_type;
+    template <typename T>
+    constexpr auto operator()(T&& arg) const -> decltype(~std::forward<T>(arg))
+    {
+        return ~std::forward<T>(arg);
+    }
+};
+
+} // namespace std
+
+#endif // __CC_ARM
+
+namespace mstd {
+using std::reference_wrapper;
+using std::ref;
+using std::cref;
+using std::plus;
+using std::minus;
+using std::multiplies;
+using std::divides;
+using std::modulus;
+using std::negate;
+using std::equal_to;
+using std::not_equal_to;
+using std::greater;
+using std::less;
+using std::greater_equal;
+using std::less_equal;
+using std::logical_and;
+using std::logical_or;
+using std::logical_not;
+using std::bit_and;
+using std::bit_or;
+using std::bit_xor;
+using std::bit_not;
+
+#if __cpp_lib_not_fn >= 201603
+using std::not_fn;
+#else
+namespace impl {
+// [func.not_fn]
+template <typename F>
+class not_fn_t {
+    std::decay_t<F> fn;
+public:
+    explicit not_fn_t(F&& f) : fn(std::forward<F>(f)) { }
+    not_fn_t(const not_fn_t &other) = default;
+    not_fn_t(not_fn_t &&other) = default;
+
+    template<typename... Args>
+    auto operator()(Args&&... args) & -> decltype(!std::declval<invoke_result_t<std::decay_t<F> &, Args...>>())
+    {
+        return !mstd::invoke(fn, std::forward<Args>(args)...);
+    }
+
+    template<typename... Args>
+    auto operator()(Args&&... args) const & -> decltype(!std::declval<invoke_result_t<std::decay_t<F> const &, Args...>>())
+    {
+        return !mstd::invoke(fn, std::forward<Args>(args)...);
+    }
+
+    template<typename... Args>
+    auto operator()(Args&&... args) && -> decltype(!std::declval<invoke_result_t<std::decay_t<F>, Args...>>())
+    {
+        return !mstd::invoke(std::move(fn), std::forward<Args>(args)...);
+    }
+
+    template<typename... Args>
+    auto operator()(Args&&... args) const && -> decltype(!std::declval<invoke_result_t<std::decay_t<F> const, Args...>>())
+    {
+        return !mstd::invoke(std::move(fn), std::forward<Args>(args)...);
+    }
+};
+}
+
+template <typename F>
+impl::not_fn_t<F> not_fn_t(F&& f)
```

Comment:
Do you mean `not_fn` as the function name ?

Reply:
Spot the person rushing to get this in before holidays. (To ensure we don't have a breaking change versus a shipped mbed_cxxsupport.h)

To some extent this is justified for the bulk of the code being for ARMC5 only, and it's not-officially-supported status, but yes, testing clearly needed.